pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+    pub fn unsafe_add_auth_for_paying_key(
+        from: IdentityId,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> u64 {
+        let auth_id = <Identity<T>>::add_auth(
+            from,
+            Signatory::Account(user_key.clone()),
+            AuthorizationData::AddRelayerPayingKey(
+                user_key.clone(),
+                paying_key.clone(),
+                polyx_limit.into(),
+            ),
+            None,
+        );
+        Self::deposit_event(RawEvent::AuthorizedPayingKey(
+            from.for_event(),
+            user_key,
+            paying_key,
+            polyx_limit.into(),
+            auth_id,
+        ));
+        auth_id
+    }
+
+    /// Check if the `key` has a valid CDD.
+    fn key_has_valid_cdd(key: &T::AccountId) -> bool {
+        if let Some(did) = <Identity<T>>::get_identity(key) {
+            <Identity<T>>::has_valid_cdd(did)
+        } else {
+            false
+        }
+    }
+
+    /// Ensure that `paying_key` is the paying key for `user_key`.
+    fn ensure_is_paying_key(user_key: &T::AccountId, paying_key: &T::AccountId) -> DispatchResult {
+        // Check if the current paying key matches
+        match <Subsidies<T>>::get(user_key) {
+            // There was no subsidy.
+            None => fail!(Error::<T>::NoPayingKey),
+            // paying key doesn't match
+            Some(s) if s.paying_key != *paying_key => fail!(Error::<T>::NotPayingKey),
+            Some(_) => Ok(()),
+        }
+    }
+
+    /// Validate and accept a `paying_key` for the `user_key`.
+    fn auth_accept_paying_key(
+        signer: Signatory<T::AccountId>,
+        from: IdentityId,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> DispatchResult {
+        // Check `signer` is DID/Key of `user_key`
+        ensure!(
+            match signer {
+                Signatory::Account(signer_key) => (signer_key == user_key),
+                Signatory::Identity(signer_did) => {
+                    <Identity<T>>::get_identity(&user_key) == Some(signer_did)
+                }
+            },
+            Error::<T>::NotAuthorizedForUserKey
+        );
+
+        // ensure that the authorization came from the DID of the paying_key.
+        ensure!(
+            <Identity<T>>::get_identity(&paying_key) == Some(from),
+            Error::<T>::NotAuthorizedForPayingKey
+        );
+
+        // ensure the user_key doesn't already have a paying_key.
+        ensure!(
+            !<Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::AlreadyHasPayingKey
+        );
+
+        // ensure both user_key and paying_key have valid CDD.
+        ensure!(
+            Self::key_has_valid_cdd(&user_key),
+            Error::<T>::UserKeyCddMissing
+        );
+        ensure!(
+            Self::key_has_valid_cdd(&paying_key),
+            Error::<T>::PayingKeyCddMissing
+        );
```

Comment:
Could you add test coverage for these branches?

Reply:
Improvements to the tests will be done in MESH-1636.

Reply:
SGTM